tensorflow/core/kernels/data/batch_dataset_op.cc

```diff
@@ -71,8 +72,11 @@ class BatchDatasetOp::Dataset : public DatasetBase {
 
   std::unique_ptr<IteratorBase> MakeIteratorInternal(
       const string& prefix) const override {
+    name_utils::IteratorPrefixParams params;
+    params.op_version = op_version_;
+    params.prefix = prefix;
     return absl::make_unique<Iterator>(Iterator::Params{
-        this, name_utils::IteratorPrefix(kDatasetType, prefix)});
+        this, name_utils::IteratorPrefix(kDatasetType, params)});
```

Comment:
Note that the iterator prefix will be different from before. For `BatchDatasetV2`, it will be `PREFIX::BatchV2`; For `BatchDataset`, it returns `PREFIX::Batch`. Before, it always returns `PREFIX::Batch`.

Reply:
Sounds good.